plugins/kosync.koplugin/main.lua

```diff
@@ -105,6 +105,18 @@ local function validateUser(user, pass)
     end
 end
 
+function KOSync:init()
+    local ReaderGesture = require("apps/reader/modules/readergesture")
+    ReaderGesture.KOSyncPushProgress = function()
+        if not self.kosync_userkey then return end
+        self:updateProgress(true)
+    end
+    ReaderGesture.KOSyncPullProgress = function()
+        if not self.kosync_userkey then return end
+        self:getProgress(true)
+    end
+end
+
```

Comment:
It might be better to use an event here?
something like (untested)
```lua
function KOSync:onKOSyncPushProgress()
    if not self.kosync_userkey then return end
    self:updateProgress(true)
end
```
called via
```lua
self.ui:handleEvent(Event:new("KOSyncPushProgress"))
```
for each of push and pull

Reply:
Thanks for suggestion but my code base on that PR: #5167 and #6035.
Should I change all to use events instead solutions with ReaderGesture.someFunction()?

Reply:
If you want, sure. It's probably not a very big deal. What do you think @poire-z ?

Reply:
Please use events, at least for new code.
It makes the code easier to read.
Also I am in middle of putting together a way to dispatch many events at once 
(to allow for profiles and gestures that do 2 things) which will only work with events.
Thanks

Reply:
Right, I was just talking about the existing code.

Reply:
Ok. Done.